packages/core-suggest/core-suggest.js

```diff
@@ -268,15 +268,27 @@ function onMutation (self) {
 
 /**
  * Handle input event in connected input
- * Performs filtering of core-suggest items
- * Dispatches event
- *  * `suggest.filter`
  * @param {CoreSuggest} self Core suggest element
  * @param {InputEvent} event
  * @returns {void}
  */
 function onInput (self, event) {
-  if (event.target !== self.input || !dispatchEvent(self, 'suggest.filter') || onAjax(self)) return
+  if (event.target !== self.input) return
+  filterItemsByInput(self)
+}
+
+/**
+ * Performs filtering of core-suggest items
+ * Dispatches event
+ *  * `suggest.filter`
+ * Filtering is aborted if
+ *  * event.preventDefault() is called on on `suggest.filter` event
+ *  * core-suggest element has attribute `ajax`
+ * @param {CoreSuggest} self Core suggest element
+ * @fires `suggest.filter`
+ */
+function filterItemsByInput (self) {
+  if (!dispatchEvent(self, 'suggest.filter') || onAjax(self)) return
```

Comment:
invoking `onAjax` on select might feel a bit overkill, but is at the same time the most coherent with present behavior.

An option would be to move this back into the `onInput` early escape, to only invoke on user input, not on user selection. Thoughts?

Reply:
[01f511a](https://github.com/nrkno/core-components/pull/656/commits/01f511afbef8f6a4244961efe2f59ec1f3afb936) introduces the latter.